### PR TITLE
Add Workflow and Command Line Tool templates during file creation.

### PR DESCRIPTION
**Pull request by ivanbatic**
When creating a new file, CLT and WF can now be prepopulated from templates.

---

This change is [Reviewable](https://reviewable.io/reviews/rabix/cottontail/55)

### CODE REVIEW

File: client/src/app/components/menu/menu-item.ts [hunk 1]
```diff
@@ -11,16 +11,16 @@ export class MenuItem {
     public readonly children: MenuItem[];
 
     constructor(name: string,
-                streams?: MenuItemUpdates = {},
-                children?: MenuItem[] = []) {
```
Comment (mayacoda) [inline client/src/app/components/menu/menu-item.ts:15]: Is there any particular reason for removing defaults and setting values with `|| []`?

File: client/src/app/components/modal/modal.component.ts [hunk 2]
```diff
@@ -47,13 +47,22 @@ export class ModalComponent {
     @ViewChild("container", {read: ViewContainerRef})
     private modalWindow: ViewContainerRef;
 
+    /** Should the modal clouse when you click on the area outside of it? */
     public closeOnOutsideClick: boolean;
+
+    /** Title of the modal window */
     public title: BehaviorSubject<string>;
+
+    /** When you press the "ESC" key, should the modal be closed? */
```
Comment (mayacoda) [inline client/src/app/components/modal/modal.component.ts:56]: Yes.

File: server/src/services/store/store.service.js [hunk 3]
```diff
@@ -325,33 +326,43 @@ module.exports = {
...
+            if (!fileDoesntExist) {
+                return deferred.reject({
                     message: 'Cannot overwrite existing file',
                     status: 403
                 });
             }
+
+            const writeFn = (filePath, content) => _self._writeFile(filePath, content).then(
+                success => deferred.resolve(success),
+                err => deferred.reject({message: err, status: 500}));
```
Comment (mayacoda) [inline server/src/services/store/store.service.js:355]: In what case would this happen?

File: server/src/services/store/templates/draft-2-command-line-tool.hbs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+    "id": "{{ id }}",
+    "class": "CommandLineTool",
+    "label": "{{ label }}",
+    "description": "{{ description }}",
```
Comment (mayacoda) [inline server/src/services/store/templates/draft-2-command-line-tool.hbs:5]: Not sure how useful interpolating description would be for templates. The user would have to enter it when creating a new file, but descriptions are supposed to be lengthy. 

The way SB handles the label is by asking the user for the name/title/label of the tool and then slugifying the name to create the unique id (in this case that would be the file name)

File: client/src/app/components/menu/menu-item.ts (hunk 1 above)
Comment (ivanbatic) [inline client/src/app/components/menu/menu-item.ts:15]: Yes, you can't assign a default value to an optional argument.
It's either `arr?: Array` or `arr: Array = []`.
I picked the first one without any particular technical reason.

File: client/src/app/components/modal/modal.component.ts (hunk 2 above)
Comment (ivanbatic) [inline client/src/app/components/modal/modal.component.ts:56]: It is a default behaviour.

File: server/src/services/store/store.service.js (hunk 3 above)
Comment (ivanbatic) [inline server/src/services/store/store.service.js:355]: For example, you might not have write permissions.

File: server/src/services/store/templates/draft-2-command-line-tool.hbs (hunk 4 above)
Comment (ivanbatic) [inline server/src/services/store/templates/draft-2-command-line-tool.hbs:5]: Isn't the id some unique RDF identifier?
Regarding description, there's a description input field in the file creation form. If it's supposed to be lengthy, we can either remove it from the form, or make it a larger text area. Depends on the actual importance of it being filled out.

Comment (mayacoda) [inline server/src/services/store/templates/draft-2-command-line-tool.hbs:5]: All the spec says about the id is literally that it's `The unique identifier for this process object.` It isn't even a required field. The ideal id for local development might just be the absolute path of the file.

It is important to fill out the description, it should be removed from the new file form and be given its own dedicated space in the UI. Moreover, the practice thus far has been that the description can handle Markdown, so it ought to have a fancy editor.

**Pull request merged**